source/extensions/path/uri_template_lib/uri_template_internal.cc

```diff
@@ -44,7 +44,7 @@ constexpr absl::string_view kLiteral = "a-zA-Z0-9-._~" // Unreserved
                                        "%"             // pct-encoded
                                        "!$&'()+,;"     // sub-delims excluding *=
                                        ":@"
-                                       "="; // user included "=" allowed
+                                       "=*"; // restricted characters
```

Comment:
cc @RyanTheOptimist as an original codeowner who reviewed the exclusion of `*` reserved character. do you have any context why it's the only char excluded? 

Using the literal `*` in a path seems really a bad practice to me however (e.g. [google search doesn't support non encoded reserved chars, not limited to *](https://developers.google.com/search/docs/crawling-indexing/url-structure)), so i am not sure if we want to expand this direction.